### ISSUE TITLE
Mistral.rs & localhost

### ISSUE THREAD

**Issue by andrebaerlocher** [caprover/caprover#2169]
**IMPORTANT:**
Please quickly search the existing issues and make sure that you're not create a duplicate issue. If you have a quesetion about a particular Dockerfile and deployment, change your questions to Dockerfile and post it where appropriate such as StackOverflow. This is a Docker question, not a CapRover specific issue.

--------------------------------------------------------------------------

Please clearly state your question:
I have two questions:
Regarding **Mistral.rs**:
I built it from the repo. Now it loops through an error:

```
2024-10-20T09:24:46.751634842Z error: 'mistralrs-server' requires a subcommand but one was not provided
2024-10-20T09:24:46.751656675Z [subcommands: toml, plain, x-lora, lora, gguf, x-lora-gguf, lora-gguf, ggml, x-lora-ggml, lora-ggml, vision-plain, diffusion-plain, help]
2024-10-20T09:24:46.751658758Z
2024-10-20T09:24:46.751659925Z Usage: mistralrs-server [OPTIONS] <COMMAND>
2024-10-20T09:24:46.751661217Z
2024-10-20T09:24:46.751662175Z For more information, try '--help'.
```
And this repeats every few seconds.

Regarding **localhost**:
I have a few services running on localhost. I tried both nginx, but failed miserably. How can I serve e.g. localhost:8080 as via CapRover?

Cheers in advance!

**Comment by githubsaturn** [comment]
@andrebaerlocher  - Mistral.rs issue is not really a CapRover question. You should reach out to Mistral.rs developers to see what the fix is. Likely, you need to do something like this:
```
FROM <your-existing-mistral.rs-image>
CMD -i toml -f toml-selectors/anymoe_lora.toml
# or something like that... but it is really a mistral.rs question.
``` 

> I have a few services running on localhost. I tried both nginx, but failed miserably. How can I serve e.g. localhost:8080 as via CapRover?

Not sure if I understand this. Did you deploy services on your server via CapRover and you're using port mapping? Or these services are created outside of CapRover?

**Comment by andrebaerlocher** [comment]
Thank you for your answer, appreciate it!

Will research more about mistral.rs elsewhere, then, cheers!

The other service is created outside of CapRover. Can I use nginx reverse proxy to access it?

**Comment by githubsaturn** [comment]
> The other service is created outside of CapRover. Can I use nginx reverse proxy to access it?

Yes, but depending on how that service is exposed. The nginx proxy is from within a container. This means that if your service (outside of CapRover) is bound to `localhost:8080` - the container cannot bind to that port. But if it's bound to `192.168.0.123:8080` for example, you can use CapRover to reverse proxy into that. 

https://stackoverflow.com/questions/20778771/what-is-the-difference-between-0-0-0-0-127-0-0-1-and-localhost
